client/app/components/sections/Topbar/Topbar.js

```diff
@@ -92,6 +95,55 @@ const DEFAULT_CONTEXT = {
   loggedInUsername: null,
 };
 
+const parseQuery = (searchQuery) => {
```

Comment:
See if the [URL](https://developer.mozilla.org/en/docs/Web/API/URL) API could be used for parsing.